extensions/wikia/AdEngine/js/lookup/prebid/adapters/veles.js

```diff
@@ -0,0 +1,163 @@
+/*global define*/
+define('ext.wikia.adEngine.lookup.prebid.adapters.veles', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.wrappers.prebid',
+	'ext.wikia.adEngine.video.vastUrlBuilder',
+	'wikia.geo',
+	'wikia.instantGlobals',
+	'wikia.window'
+], function (adContext, prebid, vastUrlBuilder, geo, instantGlobals, win) {
+	'use strict';
+
+	var bidderName = 'veles',
+		slots = {
+			// Order of slots is important - first slot name in group will be used to create ad unit
+			oasis: {
+				INCONTENT_PLAYER: {
+					sizes: [
+						[ 640, 480 ]
+					]
+				},
+				INCONTENT_LEADERBOARD: {
+					sizes: [
+						[ 640, 480 ]
+					]
+				}
+			},
+			mercury: {
+				MOBILE_IN_CONTENT: {
+					sizes: [
+						[ 640, 480 ]
+					]
+				}
+			}
+		};
+
+	function parseParameters(adParameters) {
+		var parameters = {};
+
+		if (adParameters.childNodes.length && adParameters.childNodes[0].nodeValue) {
+			adParameters.childNodes[0].nodeValue.split(',').forEach(function (pair) {
+				var data = pair.split('=');
+
+				parameters[data[0]] = data[1];
+			});
+		}
+
+		return parameters;
```

Comment:
Are you sure `parameters` won't be always `{}`? My memory isn't always perfect but I wonder if `forEach` isn't asynchronous.